client/landing/gutenboarding/components/domain-picker/button.tsx

```diff
@@ -60,6 +64,11 @@ const DomainPickerButton: FunctionComponent = () => {
 		[ search, siteVertical ]
 	);
 
+	const handleDomainSelect: typeof onDomainSelect = selectedDomain => {
+		setDomainPopoverVisibility( false );
+		onDomainSelect( selectedDomain );
```

Comment:
Since we have this component already using hooks to connect to the store, why don't we also dispatch `setDomain` here to keep all the domain related logic in one place?

Reply:
I'm thinking about the domain picker as a component independent of onboarding.

https://github.com/Automattic/wp-calypso/pull/37820 makes this clearer. I think we should move away from depending on the onboarding store in the domain-picker.